s3-handler/main.go

```diff
@@ -182,6 +234,44 @@ func initSamplingRules() error {
 	return err
 }
 
+func parseLineFilterRules(input []byte) ([]lineFilterRule, error) {
+	var result []lineFilterRule
+	err := json.Unmarshal(input, &result)
+	return result, err
+}
+
+func initLineFilterRules() error {
+	rulesRaw := os.Getenv("LINE_FILTER_RULES")
+
+	if rulesRaw == "" {
+		return nil
+	}
+	logrus.WithField("LINE_FILTER_RULES", rulesRaw).Info("Got line filter rules")
+
+	var err error
+	lineFilterRules, err = parseLineFilterRules([]byte(rulesRaw))
+	if len(lineFilterRules) < 1 {
```

Comment:
Need to check the returned err and the len checks can't be negative, so it's easier to understand with a equal 0 check.

```suggestion
	lineFilterRules, err := parseLineFilterRules([]byte(rulesRaw))
	if err != nil {
		return err
	}
	if len(lineFilterRules) == 0 {
```